gobblin-data-management/src/test/java/org/apache/gobblin/util/commit/SetPermissionCommitStepTest.java

```diff
@@ -72,11 +62,47 @@ public void tearDown() throws IOException {
   }
 
   @Test
-  public void testExecute() throws IOException {
-    this.step.execute();
+  public void testExecuteSingleLevel() throws IOException {
+    Path dir1 = new Path(ROOT_DIR, "dir1");
+    String owner = this.fs.getFileStatus(ROOT_DIR).getOwner();
+    String group = this.fs.getFileStatus(ROOT_DIR).getGroup();
+    FsPermission permission = new FsPermission(FsAction.ALL, FsAction.ALL, FsAction.ALL);
+    fs.mkdirs(dir1);
+    OwnerAndPermission ownerAndPermission = new OwnerAndPermission(owner, group, permission);
+    Map<String, OwnerAndPermission> pathAndPermissions = new HashMap<>();
+    pathAndPermissions.put(dir1.toString(), ownerAndPermission);
+
+    CommitStep step = new SetPermissionCommitStep(this.fs, pathAndPermissions, new Properties());
+    step.execute();
     Assert.assertEquals(this.fs.exists(dir1), true);
     Assert.assertEquals(this.fs.getFileStatus(dir1).getPermission(), permission);
     Assert.assertEquals(this.fs.getFileStatus(dir1).getOwner(), this.fs.getFileStatus(ROOT_DIR).getOwner());
     Assert.assertEquals(this.fs.getFileStatus(dir1).getGroup(), this.fs.getFileStatus(ROOT_DIR).getGroup());
   }
+
+  @Test
+  public void testExecuteNested() throws IOException {
+    Path dirNestedParent = new Path(ROOT_DIR, "dirParent");
+    Path dirNestedChild = new Path(ROOT_DIR, "dirParent/dirChild");
+    String owner = this.fs.getFileStatus(ROOT_DIR).getOwner();
+    String group = this.fs.getFileStatus(ROOT_DIR).getGroup();
+    FsPermission permissionParent = new FsPermission(FsAction.ALL, FsAction.ALL, FsAction.ALL);
+    OwnerAndPermission ownerAndPermissionParent = new OwnerAndPermission(owner, group, permissionParent);
+    fs.mkdirs(dirNestedChild);
+    FsPermission permissionChild = new FsPermission(FsAction.ALL, FsAction.NONE, FsAction.NONE);
+    OwnerAndPermission ownerAndPermissionChild = new OwnerAndPermission(owner, group, permissionChild);
+
+    Map<String, OwnerAndPermission> pathAndPermissions = new TreeMap<>();
+    pathAndPermissions.put(dirNestedParent.toString(), ownerAndPermissionParent);
+    pathAndPermissions.put(dirNestedChild.toString(), ownerAndPermissionChild);
```

Comment:
curious, why use a `TreeMap` here?  if it's ordering guarantees are necessary for `SetPermissionCommitStep`, that should be the second param's type.  the test should validate they're unnecessary by constructing a `HashMap`